src/main/java/ogc/rs/database/DatabaseServiceImpl.java

```diff
@@ -74,18 +92,53 @@ public Future<List<JsonObject>> getCollections() {
                         " from collections_details join collection_supported_crs" +
                         " on collections_details.id = collection_supported_crs.collection_id join crs_to_srid" +
                         " on crs_to_srid.id = collection_supported_crs.crs_id join collection_type" +
-                        " on collections_details.id = collection_type.collection_id group by collections_details.id")
+                        " on collections_details.id = collection_type.collection_id where collection_type.type!= 'STAC' group by collections_details.id")
                     .collecting(collector)
                     .execute()
-                    .map(SqlResult::value))
+                    .map(SqlResult::value)
             .onSuccess(success -> {
-                LOGGER.debug("Collections Result: {}", success.toString());
-                result.complete(success);
+                if (success.isEmpty()) {
+                    LOGGER.error("Collections table is empty!");
+                    result.fail(
+                            new OgcException(404, "Not found", "Collection table is Empty!"));
+                } else {
+                    conn.preparedQuery("SELECT * FROM COLLECTIONS_ENCLOSURE")
+                            .collecting(collector)
+                            .execute()
+                            .map(SqlResult::value)
+                            .onSuccess(
+                                    enclosureResult -> {
+                                        if (enclosureResult.isEmpty()) {
+                                            LOGGER.warn("Assets table is empty!");
+                                            result.complete(success);
+                                        } else {
+                                            for (JsonObject enclosure : enclosureResult) {
```

Comment:
Instead of running a double for loop, what you can try and verify is to change the query itself to join the table `collections_enclosure` and use `json_agg()`, refer- https://github.com/datakaveri/ogc-resource-server/blob/7e75e1a37e2e51ddeb0801f1622eb77cf46f4f30/src/main/java/ogc/rs/database/DatabaseServiceImpl.java#L417